packages/Discretization/test/tstTpetraVectorView.cpp

```diff
@@ -0,0 +1,163 @@
+
+//---------------------------------------------------------------------------//
+// Test building and manipulating a tpetra vector with a kokkos view.
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( TpetraVectorView, vector_view, Node )
+{
+    // Template aliases.
+    using SC = double;
+    using LO = int;
+    using GO = unsigned long int;
+    using NO = Node;
+
+    // Get the communicator.
+    auto comm = Teuchos::DefaultComm<int>::getComm();
+
+    // Set the number of global elements.
+    LO num_local = 100;
+    GO num_global = num_local * comm->getSize();
+
+    // Create a map.
+    auto map =
+        Tpetra::createUniformContigMapWithNode<LO, GO, NO>( num_global, comm );
+
+    // Type aliases.
+    using DualViewType =
+        typename Tpetra::MultiVector<SC, LO, GO, NO>::dual_view_type;
+    using DeviceType = typename DualViewType::device_type;
+    using ExecutionSpace = typename DualViewType::execution_space;
+
+    // Create a view and get the device view.
+    int num_vec = 3;
+    DualViewType dual_view( "test_dual_view", num_local, num_vec );
+    auto dev_view = dual_view.template view<DeviceType>();
+
+    // Put some data in the view.
+    SC value = 3.3;
+    auto fill_func = KOKKOS_LAMBDA( int i )
+    {
+        for ( int d = 0; d < num_vec; ++d )
+            dev_view( i, d ) = value;
+    };
+    Kokkos::parallel_for( REGION_NAME( "fill_test_dual_vew" ),
+                          Kokkos::RangePolicy<ExecutionSpace>( 0, num_local ),
+                          fill_func );
+
+    // Create a vector from the view.
+    auto vec = Teuchos::rcp(
+        new Tpetra::MultiVector<SC, LO, GO, NO>( map, dual_view ) );
+
+    // Check the vector norm.
+    Kokkos::View<SC *, DeviceType> norms( "vector norms", num_vec );
+    SC test_norm = num_global * value;
+    vec->norm1( norms );
+    Kokkos::View<SC *, Kokkos::HostSpace> host_norms( "host vector norms",
+                                                      num_vec );
+    Kokkos::deep_copy( host_norms, norms );
+    for ( int d = 0; d < num_vec; ++d )
+        TEST_FLOATING_EQUALITY( test_norm, host_norms( d ), 1.0e-14 );
+}
+
+//---------------------------------------------------------------------------//
+// Test converting between a regular view and a dual view.
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( TpetraVectorView, dual_view, Node )
+{
+    // Template aliases.
+    using SC = double;
+    using LO = int;
+    using GO = unsigned long int;
+    using NO = Node;
+
+    // Type aliases.
+    using DualViewType =
+        typename Tpetra::MultiVector<SC, LO, GO, NO>::dual_view_type;
+    using DeviceType = typename DualViewType::device_type;
+    using ExecutionSpace = typename DualViewType::execution_space;
+
+    // Create a dual view.
+    LO num_local = 10;
+    int num_vec = 3;
+    DualViewType dual_view( "test_dual_view", num_local, num_vec );
+
+    // Get a regular view on the device.
+    auto device_view = dual_view.template view<DeviceType>();
```

Comment:
I am not sure I follow why you use `dual_view` and `new_dual_view` in this unit test.

Please make it clear that you are wrapping a regular View into a DualView.
```C++
// define some kind of function that wraps a regular View into a DualView
DualView toDualView( Kokkos::View<SC *, DeviceType> const & in )
{ ... }

// then in your unit test
TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( ... )
{
// create a regular View as we usually do
Kokkos::View<SC *, DeviceType> v ( "v", 2  );
// initialize it to whatever
Kokkos::parallel_for( REGION_NAME( "iota" ),
                      Kokkos::RangePolicy<ExecutionSpace>( 0, num_local ),
                      DataTransferKit::Iota<DeviceType>( v ) );

// wrap it into a DualView
auto d = toDualView( v );

// inspect d to make sure it is initialized correctly
...

// change the dual view and check that v changed as well
...

// change v and look at d
...
}
```